### PR TITLE
parser: let parser support `flashback cluster`

### LINKED ISSUE #37229: Parser supports `flash back cluster as of timestamp 'expression'`

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #37229

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support `flashback cluster` stmt
```

### CODE REVIEW

File: parser/ast/ddl.go [hunk 1]
```diff
@@ -4038,6 +4038,42 @@ func (n *RecoverTableStmt) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+// FlashBackClusterStmt is a statement to restore the cluster to the specified timestamp
+type FlashBackClusterStmt struct {
+	ddlNode
+
+	AsOf *AsOfClause
```
Comment (tangenta) [inline parser/ast/ddl.go:4045]: If the `AsOfClause` is required, why not making it a value instead of a pointer?

Comment (xhebox) [reply parser/ast/ddl.go:4045]: +1

Comment (Defined2014) [reply parser/ast/ddl.go]: updated, PTAL @xhebox @tangenta

**Review by xhebox — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**